Description: <short summary of the patch>
Author: Miao Changwei <miaochangwei@uniontech.com>

---
Last-Update: 2022-04-28

--- systemtap-4.5.orig/config.guess
+++ systemtap-4.5/config.guess
@@ -1014,6 +1014,9 @@ EOF
     sparc:Linux:*:* | sparc64:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-${LIBC}
 	exit ;;
+    sw_64:Linux:*:*)
+	echo ${UNAME_MACHINE}-unknown-linux-${LIBC}
+	exit ;;
     tile*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-${LIBC}
 	exit ;;
--- systemtap-4.5.orig/config.sub
+++ systemtap-4.5/config.sub
@@ -310,6 +310,7 @@ case $basic_machine in
 	| sparc | sparc64 | sparc64b | sparc64v | sparc86x | sparclet | sparclite \
 	| sparcv8 | sparcv9 | sparcv9b | sparcv9v \
 	| spu \
+	| sw_64 \
 	| tahoe | tic4x | tic54x | tic55x | tic6x | tic80 | tron \
 	| ubicom32 \
 	| v850 | v850e | v850e1 | v850e2 | v850es | v850e2v3 \
--- systemtap-4.5.orig/staplog.c
+++ systemtap-4.5/staplog.c
@@ -24,7 +24,7 @@
    X86, X86_64, etc.  See crash/configure.c (!). */
 #if !defined(X86) && !defined(X86_64) && !defined(ALPHA) && !defined(PPC) && \
     !defined(IA64) && !defined(PPC64) && !defined(S390) && !defined(S390X) && \
-    !defined(ARM) && !defined(ARM64)
+    !defined(ARM) && !defined(ARM64) && !defined(SW64)
 
 #if defined(__alpha__)
 #define ALPHA
@@ -46,6 +46,8 @@
 #define ARM64
 #elif defined(__arm__)
 #define ARM
+#elif defined(__sw_64__)
+#define SW64
 #else
 #warn "unknown architecture for crash/staplog support"
 #endif
